edx_event_bus_kafka/internal/consumer.py

```diff
@@ -381,6 +370,31 @@ def emit_signals_from_message(self, msg):
         if AUDIT_LOGGING_ENABLED.is_enabled():
             logger.info('Message from Kafka processed successfully')
 
+    def deserialize_message_value(self, msg):
+        """
+        Deserialize an Avro message value
+
+        Arguments:
+            msg (Message): the raw message from the consumer
+
+        Returns:
+            The deserialized message value
+        """
+        schema_registry_client = get_schema_registry_client()
+
+        signal_deserializer = AvroSignalDeserializer(self.signal)
+
+        def inner_from_dict(event_data_dict, ctx=None):  # pylint: disable=unused-argument
+            return signal_deserializer.from_dict(event_data_dict)
+
+        # We do not deserialize the key because we don't need it for anything yet.
+        # Also see https://github.com/openedx/openedx-events/issues/86 for some challenges on determining key schema.
+        avro_deserializer = AvroDeserializer(schema_str=signal_deserializer.schema_string(),
+                                             schema_registry_client=schema_registry_client,
+                                             from_dict=inner_from_dict)
```

Comment:
Can we move `avro_deserializer` and the lines above it to a cached function? See `get_serializers` in the producer code, including the `_reset_caches` function at the end of the file. If I recall correctly, creating the deserializer isn't _terribly_ computationally intensive, but far more so than a hash lookup.